ci/generate-markdown/update-library-landing-dox.sh

```diff
@@ -52,10 +53,43 @@ while IFS= read -r -d $'\0' option_defaults_cc; do
   env_vars+=("  \`EndpointOption\` (which defaults to ${endpoint})")
   env_vars+=("  used by \`${make_connection}()\`.")
   env_vars+=("")
-done < <(git ls-files -z -- "${lib}/internal/*_option_defaults.cc")
-sed -i -f - "${lib}/doc/main.dox" <<EOT
+done < <(git ls-files -z -- "${LIB}/internal/*_option_defaults.cc")
+
+sed -i -f - "${MAIN_DOX}" <<EOT
 /${inject_start}/,/${inject_end}/c \\
 ${inject_start}\\
 $(printf '%s\\\n' "${env_vars[@]}")
 ${inject_end}
 EOT
+
+IFS= mapfile -d $'\0' -t samples_cc < <(git ls-files -z -- "${LIB}/samples/*_client_samples.cc")
+
+(
+  sed '/<!-- inject-endpoint-snippet-start -->/q' "${MAIN_DOX}"
+  if [[ ${#samples_cc[@]} -gt 0 ]]; then
+    sample_cc="${samples_cc[0]}"
+    client_name="$(sed -n '/main-dox-marker: / s;// main-dox-marker: \(.*\);\1;p' "${sample_cc}")"
+    echo 'For example, this will override the default endpoint for `'"${client_name}"'`:'
+    echo
+    echo "@snippet $(basename "${sample_cc}") set-client-endpoint"
+    if [[ ${#samples_cc[@]} -gt 1 ]]; then
+      echo
+      echo "Follow these links to find examples for other \\c *Client classes:"
+      for sample_cc in "${samples_cc[@]}"; do
+        sed -n "/main-dox-marker: / s;// main-dox-marker: \(.*\); [\1](@ref \1-endpoint-snippet);p" "${sample_cc}"
+      done
+    fi
+  fi
+  echo
+  sed -n '/<!-- inject-endpoint-snippet-end -->/,$p' "${MAIN_DOX}"
+) | sponge "${MAIN_DOX}"
```

Comment:
I might suggest that these could also be handled with `sed -i`, but that is a question for another day.

Reply:
Ack.